src/vs/workbench/contrib/debug/browser/debugConfigurationManager.ts

```diff
@@ -251,23 +253,50 @@ export class ConfigurationManager implements IConfigurationManager {
 	async getDynamicProviders(): Promise<{ label: string, pick: () => Promise<{ launch: ILaunch, config: IConfig } | undefined> }[]> {
 		const extensions = await this.extensionService.getExtensions();
 		const onDebugDynamicConfigurationsName = 'onDebugDynamicConfigurations';
-		const debugDynamicExtensionsTypes = extensions.map(e => {
-			const activationEvent = e.activationEvents && e.activationEvents.find(e => e.includes(onDebugDynamicConfigurationsName));
-			if (activationEvent) {
-				const type = activationEvent.substr(onDebugDynamicConfigurationsName.length + 1);
-				return type || (e.contributes && e.contributes.debuggers && e.contributes.debuggers.length ? e.contributes.debuggers[0].type : undefined);
+		const debugDynamicExtensionsTypes = extensions.reduce((acc, e) => {
+			if (!e.activationEvents) {
+				return acc;
 			}
 
-			return undefined;
-		}).filter(type => typeof type === 'string' && !!this.getDebuggerLabel(type)) as string[];
+			const explicitTypes = e.activationEvents.filter(e => e.includes(`${onDebugDynamicConfigurationsName}:`)).map(type => type.slice(onDebugDynamicConfigurationsName.length + 1));
+			if (explicitTypes.length) {
+				return [...acc, ...explicitTypes];
+			}
+
+			if (e.activationEvents.includes(onDebugDynamicConfigurationsName)) {
+				const debuggerType = e.contributes?.debuggers?.[0].type;
+				return debuggerType ? [...acc, debuggerType] : acc;
+			}
+
+			return acc;
+		}, [] as string[]);
 
 		return debugDynamicExtensionsTypes.map(type => {
 			return {
 				label: this.getDebuggerLabel(type)!,
 				pick: async () => {
+					const input = this.quickInputService.createQuickPick<IDynamicPickItem>();
```

Comment:
Yeah I personally dislike the `createQuickPick` API since it brings too much noisy compared to the previous `.pick` usage.
Are we using the new `createQuickPick` only for the `.busy` flag or is ther emore to it

Reply:
It's so that we can create the quick pick before we start activating debuggers and getting results, and so that we have control over it to auto-select a single item if there's only one (if we just wrapped everything in a promise given to `.pick` I don't think we'd be able to do that automatic selector)

Reply:
Ok, makes sense